here-naksha-lib-hub/src/main/java/com/here/naksha/lib/hub/mock/MockResultCursor.java

```diff
@@ -0,0 +1,299 @@
+
+  private final @NotNull Class<T> featureType;
+  private final @NotNull List<Object> items;
+  protected int currentPos;
+
+  MockResultCursor(@NotNull Class<T> featureType, @NotNull List<Object> items) {
+    this.featureType = featureType;
+    this.items = items;
+    currentPos = -1;
+  }
+
+  /**
+   * Tests whether there is another result.
+   *
+   * @return {@code true} if another result is available; {@code false} otherwise.
+   */
+  @Override
+  public boolean hasNext() {
+    return currentPos < items.size() - 1;
+  }
+
+  /**
+   * Moves the cursor to the next result.
+   *
+   * @return {@code true}, if the cursor is on a valid result; {@code false} otherwise.
+   */
+  @Override
+  public boolean next() {
+    currentPos++;
+    return isPositionValid();
+  }
+
+  /**
+   * Moves the cursor to the previous result.
+   *
+   * @return {@code true}, if the cursor is on a valid result; {@code false} otherwise.
+   */
+  @Override
+  public boolean previous() {
+    currentPos--;
+    return isPositionValid();
+  }
+
+  /**
+   * Moves the cursor before the first result, so that calling {@link #next()} will load the first result.
+   */
+  @Override
+  public void beforeFirst() {
+    currentPos = -1;
+  }
+
+  /**
+   * Moves the cursor to the first result.
+   *
+   * @return {@code true}, if the cursor is on a valid result; {@code false} otherwise.
+   */
+  @Override
+  public boolean first() {
+    currentPos = 0;
+    return isPositionValid();
+  }
+
+  /**
+   * Moves the cursor behind the last result, so that calling {@link #previous()} will load the last result.
+   */
+  @Override
+  public void afterLast() {
+    currentPos = items.size();
+  }
+
+  /**
+   * Moves the cursor to the last result.
+   *
+   * @return {@code true}, if the cursor is on a valid result; {@code false} otherwise.
+   */
+  @Override
+  public boolean last() {
+    currentPos = items.size();
+    return isPositionValid();
+  }
+
+  /**
+   * Moves the cursor forward or backward from the current position by the given {@code amount}.
+   *
+   * @param amount The amount of results to move.
+   * @return {@code true}, if the cursor is on a valid result; {@code false} otherwise.
+   */
+  @Override
+  public boolean relative(long amount) {
+    currentPos += amount;
+    return isPositionValid();
+  }
+
+  /**
+   * Moves the cursor to the given position in the result-set. Setting the position to a negative number (for example to {@code -1}) has
+   * the same effect as calling {@link #beforeFirst()}. Moving to any position being behind the last result has the same effect as calling
+   * {@link #afterLast()}.
+   *
+   * @param position The absolute position with the first result being at position zero.
+   * @return {@code true}, if the cursor is on a valid result; {@code false} otherwise.
+   */
+  @Override
+  public boolean absolute(long position) {
+    currentPos = (int) position;
```

Comment:
I'm thinking maybe we should `throw new NotImplementedException();` also for this and getOp() and other methods we are not using, so we know exactly what the problem is when we later need these methods in our tests

Reply:
I would implement things that we know how implement - this one is one of them, it's simple in-memory itrator-like structure. I would opt for leaving it as it is

Reply:
that's good as well :) though you should also add in some if checks. The docs mentions that `Setting the position to a negative number (for example to {@code -1}) has
   * the same effect as calling {@link #beforeFirst()}. Moving to any position being behind the last result has the same effect as calling
   * {@link #afterLast()}.`